src/NServiceBus.Core.Tests/Sagas/ComponentTests/When_persisting_a_saga_with_the_same_unique_property_as_a_completed_saga.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace NServiceBus.Persistence.ComponentTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_persisting_a_saga_with_the_same_unique_property_as_a_completed_saga : SagaPersisterTests<SagaWithCorrelationProperty, SagaWithCorrelationPropertyData>
```

Comment:
This one is a bit questionable, can we really assume that persisters are able to delete the saga immediately? 

See https://github.com/Particular/docs.particular.net/issues/2523 for a related discussion